model versions

MichelleArk opened the pull request:
Addresses: https://github.com/dbt-labs/docs.getdbt.com/issues/3099

## What are you changing in this pull request and why?
<!---
Describe your changes and why you're making them. If linked to an open
issue or a pull request on dbt Core, then link to them here! 

To learn more about the writing conventions used in the dbt Labs docs, see the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md).
-->

## Checklist
<!--
Uncomment if you're publishing docs for a prerelease version of dbt (delete if not applicable):
- [ ] Add versioning components, as described in [Versioning Docs](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/versioningdocs.md)
- [ ] Add a note to the prerelease version [Migration Guide](https://github.com/dbt-labs/docs.getdbt.com/tree/current/website/docs/guides/migration/versions)
-->
- [ ] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) and [About versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) so my content adheres to these guidelines.
- [ ] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

Adding new pages (delete if not applicable):
- [ ] Add page to `website/sidebars.js`
- [ ] Provide a unique filename for the new page

Removing or renaming existing pages (delete if not applicable):
- [ ] Remove page from `website/sidebars.js`
- [ ] Add an entry `_redirects`
- [ ] [Ran link testing](https://github.com/dbt-labs/docs.getdbt.com#running-the-cypress-tests-locally) to update the links that point to the deleted page

MichelleArk committed:
first pass - model versions

dave-connors-3 commented on website/docs/docs/collaborate/publish/model-versions.md:
```diff
@@ -15,17 +15,71 @@ For more details and to leave your feedback, check out the GitHub discussion:
+    versions:
+      - v: 2
+        columns:
+          - include: *
```
@MichelleArk is this intended to be an array? or a dictionary with `include` and `exclude` keys (among possible others)?

MichelleArk replied:
Currently we're thinking it'll be an array element of the `columns` array. So you could do something like: 

```yaml
columns:
  - include: "*"
    exclude: column_name_to_exclude
  - name: customer_id  # included in addition the '*' set. if customer_id were in the '*' set -> override it
    description: This is the primary key
    data_type: float
  - name: country_name # included in addition the '*' set. if country_name were in the '*' set -> override it
    description: Where this customer lives
    data_type: string
```

matthewshaver commented on website/docs/docs/collaborate/publish/model-versions.md:
```diff
@@ -15,17 +15,71 @@ For more details and to leave your feedback, check out the GitHub discussion:
+
+</File>
+
+If you wanted to make a breaking change to the model - for example, removing a column - you'd create a new model file (e.g. SQL file) encompassing those breaking changes. The default convention is to name the new file would with a `_v<version>` suffix. The new version can then be configured in relation to the original model:
```
```suggestion
If you wanted to make a breaking change to the model - for example, removing a column - you'd create a new model file (for example, SQL file) encompassing those breaking changes. The default convention is naming the new file with a `_v<version>` suffix. The new version can then be configured in relation to the original model:
```

matthewshaver committed:
Update website/docs/docs/collaborate/publish/model-versions.md

jtcohen6 commented on website/docs/docs/collaborate/publish/model-versions.md:
```diff
@@ -15,17 +15,71 @@ For more details and to leave your feedback, check out the GitHub discussion:
+
+</File>
+
+The above configuration will create two models (one for each version), with aliases `dim_customers` (version 1) and `dim_customers_v2`.
```
We'll definitely want to call out that projects which have historically reimplemented `generate_alias_name` will need to update their custom implementations to account for model versions. Otherwise they'll see something like this as soon as they start using versions:
```
dbt.exceptions.AmbiguousAliasError: Compilation Error
  dbt found two resources with the database representation "database.schema.model_name".
  dbt cannot create two resources with identical database representations. To fix this,
  change the configuration of one of these resources:
  - model.project_name.model_name.v1 (models/.../model_name.sql)
  - model.project_name.model_name.v2 (models/.../model_name_v2.sql)
```
I'm not sure if it makes more sense here or in the migration guide (maybe both?)

jtcohen6 commented on website/docs/reference/model-properties.md:
```diff
@@ -32,6 +33,25 @@ models:
 
       - name: ... # declare properties of additional columns
+    versions:
+      - v: <version_name> # required
```
- Clarify that this can be string or numeric
- Somewhere we say that we'll do ordering accordingly (but we don't have any features yet that will actually use it - that will come in https://github.com/dbt-labs/dbt-core/issues/7199)

MichelleArk replied:
> but we don't have any features yet that will actually use it 

we do! calling `ref` on a versioned model without the `version` argument relies on the latest model version and ordering implementation (if `latest_version` isn't explicitly configured)

runleonarun committed:
Create release note for scheduler (#3149)

## What are you changing in this pull request and why?
Per [this
request](https://www.notion.so/dbtlabs/The-dbt-Cloud-Scheduler-will-now-process-over-scheduled-jobs-more-intelligently-bc08563a9c3d4411bbbd96d5a9a1f1e6?pvs=4),
adding a release note.

## Checklist
- [x] Review the [Content style
guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md)
and [About
versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version)
so my content adheres to these guidelines.
- [ ]  "needs technical review"

---------

Co-authored-by: mirnawong1 <89008547+mirnawong1@users.noreply.github.com>

matthewshaver committed:
Minor editorial adjustments (#3172)

## What are you changing in this pull request and why?
Minor editorial adjustments

## Checklist
<!--
Uncomment if you're publishing docs for a prerelease version of dbt
(delete if not applicable):
- [ ] Add versioning components, as described in [Versioning
Docs](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-entire-pages)
- [ ] Add a note to the prerelease version [Migration
Guide](https://github.com/dbt-labs/docs.getdbt.com/tree/current/website/docs/guides/migration/versions)
-->
- [ ] Review the [Content style
guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md)
and [About
versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version)
so my content adheres to these guidelines.
- [ ] Add a checklist item for anything that needs to happen before this
PR is merged, such as "needs technical review" or "change base branch."

Adding new pages (delete if not applicable):
- [ ] Add page to `website/sidebars.js`
- [ ] Provide a unique filename for the new page

Removing or renaming existing pages (delete if not applicable):
- [ ] Remove page from `website/sidebars.js`
- [ ] Add an entry `_redirects`
- [ ] [Ran link
testing](https://github.com/dbt-labs/docs.getdbt.com#running-the-cypress-tests-locally)
to update the links that point to the deleted page

---------

Co-authored-by: Leona B. Campbell <3880403+runleonarun@users.noreply.github.com>

matthewshaver committed:
Fixing adapter link

MichelleArk committed:
Merge branch 'current' into v1.5rc-versions

matthewshaver committed:
Fixing adapter link (#3176)

## What are you changing in this pull request and why?
<!---
Describe your changes and why you're making them. If linked to an open
issue or a pull request on dbt Core, then link to them here! 

To learn more about the writing conventions used in the dbt Labs docs,
see the [Content style
guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md).
-->

## Checklist
<!--
Uncomment if you're publishing docs for a prerelease version of dbt
(delete if not applicable):
- [ ] Add versioning components, as described in [Versioning
Docs](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-entire-pages)
- [ ] Add a note to the prerelease version [Migration
Guide](https://github.com/dbt-labs/docs.getdbt.com/tree/current/website/docs/guides/migration/versions)
-->
- [x] Review the [Content style
guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md)
and [About
versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version)
so my content adheres to these guidelines.
- [ ] Add a checklist item for anything that needs to happen before this
PR is merged, such as "needs technical review" or "change base branch."

MichelleArk committed:
defined_in, include/exclude, v

MichelleArk committed:
Merge branch 'current' into v1.5rc-versions

jtcohen6 committed:
Merge remote-tracking branch 'origin/jerco/multi-project-last-round' into v1.5rc-versions
